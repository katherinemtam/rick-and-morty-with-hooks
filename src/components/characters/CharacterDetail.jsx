import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetail = ({ name, status, species, type, gender, origin, location, image }) => (
  <figure>
    <h2>{name}</h2>
    <img src={image}/>
    <figcaption>
      <p>Status: {status}</p>
      <p>Species: {species}</p>
      <p>Type: {type ? { type } : 'N/A'}</p>
      <p>Gender: {gender}</p>
      <p>Origin: {origin.name}</p>
      <p>Location: {location.name}</p>
    </figcaption>
  </figure>
);

CharacterDetail.propTypes = {
  name: PropTypes.string,
  status: PropTypes.string,
  species: PropTypes.string,
  type: PropTypes.string,
  gender: PropTypes.string,
  origin: PropTypes.object,
  location: PropTypes.object,
  image: PropTypes.string,
};

export default CharacterDetail;
