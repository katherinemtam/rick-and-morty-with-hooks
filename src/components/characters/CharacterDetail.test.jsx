import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetail from './CharacterDetail';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterDetail component', () => {
  const character = 
    {
      name: 'test',
      status: 'test',
      species: 'test',
      type: 'test',
      gender: 'test',
      origin: {
        name: 'test'
      },
      location: {
        name: 'test'
      },
      image:'test',
    }
  ;

  afterEach(() => cleanup());
  it('renders CharacterDetail', () => {
    const { asFragment } = render(<MemoryRouter><CharacterDetail 
      name={character.name}
      status={character.status}
      species={character.species}
      type={character.type}
      gender={character.gender}
      origin={character.origin.name}
      location={character.location.name}
      image={character.image}
    /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});

