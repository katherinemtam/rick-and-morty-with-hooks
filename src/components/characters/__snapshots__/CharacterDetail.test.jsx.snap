// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterDetail component renders CharacterDetail 1`] = `
<DocumentFragment>
  <figure>
    <h2>
      test
    </h2>
    <img
      src="test"
    />
    <figcaption>
      <p>
        Status: test
      </p>
      <p>
        Species: test
      </p>
      <p>
        Type: test
      </p>
      <p>
        Gender: test
      </p>
      <p>
        Origin: test
      </p>
      <p>
        Location: test
      </p>
    </figcaption>
    <a
      href="/"
    >
      Return Home
    </a>
  </figure>
</DocumentFragment>
`;
